aztk_cli/spark/endpoints/cluster/cluster_ssh.py

```diff
@@ -88,16 +73,72 @@ def print_plugin_ports(cluster_config: ClusterConfiguration):
                 if port.expose_publicly:
                     has_ports = True
                     plugin_ports[plugin.name].append(port)
-        
+
         if has_ports:
             log.info("plugins:")
 
         for plugin in plugin_ports:
             if plugin_ports[plugin]:
-                log.info("  " + plugin)
+                log.info(" %s ", plugin)
                 for port in plugin_ports[plugin]:
                     label = "    - open"
                     if port.name:
                         label += " {}".format(port.name)
                     url = "{0}{1}".format(http_prefix, port.public_port)
                     utils.log_property(label, url)
+
+
+def native_python_ssh_into_master(spark_client, cluster, ssh_conf, password):
+    if not ssh_conf.connect:
+        log.warning("No ssh client found, using pure python connection.")
+        return
+
+    configuration = spark_client.get_cluster_config(cluster.id)
+    plugin_ports = []
+    if configuration and configuration.plugins:
+        ports = [
+            PortForwardingSpecification(
+                port.internal,
+                port.public_port) for plugin in configuration.plugins for port in plugin.ports if port.expose_publicly
+        ]
+        plugin_ports.extend(ports)
+
+    print("Press ctrl+c to exit...")
```

Comment:
shouldn't that be ctrl + d? or not as we are not giving shell access

Reply:
We aren't giving shell access, the process just hangs so the ports stay open until killed (with `crl + c`).